docs/src/index.md

```diff
@@ -134,6 +134,27 @@ Test Summary:                    | Pass  Total
 relu at -0.5, with cotangent 1.0 |    4      4
 ```
 
+## Custom finite differencing
+
+If a package is using a custom finite differencing method of testing the `frule`s and
+`rrule`s, `check_equal` function provides a convenient way of comparing [various types]
+(https://www.juliadiff.org/ChainRulesCore.jl/dev/design/many_differentials.html#Design-Notes:-The-many-to-many-relationship-between-differential-types-and-primal-types.)
+of differentials.
+
+It is effectively `isapprox` but with better error messages because it knows about `thunk`s
```

Comment:
why isn't it called `check_approx`, and/or why doesn't it extend `isapprox`?

also why is it called `check_*` rather than `test_*`?

I'm wondering if (i) there are good reasons and they should be explained on this page, or (ii) we should consider renaming this whenever we do #96

Reply:
We are trying to deprecate `isapprox` methods. I think one of the reasons is that we don't want to pirate `isapprox` for tuple arguments, but there may be other. Should we document that, or is it too low level?

I guess we could call it `test_*` as you suggest, maybe just `test_equal`?

Reply:
thanks :) i think a comment on why this is not just `isapprox` is appropriate. 

Seems `test_approx` would be a better name (i'll add it to #96)